src/eduid_webapp/email/views.py

```diff
@@ -219,28 +221,41 @@ def verify_link(user):
         db = current_app.proofing_statedb
         state = db.get_state_by_eppn_and_email(proofing_user.eppn, email, raise_on_missing=False)
 
+        url = urlparse.urljoin(current_app.config['DASHBOARD_URL'], 'emails')
+        url_parts = list(urlparse.urlparse(url))
+
         if state is None:
             current_app.logger.info("Missing state for verification code received for email {} "
                                     "and user {}.".format(email, user))
-            return redirect(current_app.config['SAML2_LOGIN_REDIRECT_URL'])
+            url_parts[4] = urlencode({'msg': ':ERROR:emails.unknown_email'})
```

Comment:
Is it possible to make the part where you use index 4 are a bit more clear?

Reply:
Done